src/crimson/os/seastore/transaction.h

```diff
@@ -92,6 +101,13 @@ class Transaction {
     write_set.insert(*ref);
   }
 
+  void add_rewrite_extent(CachedExtentRef ref) {
+    assert(ref->is_logical());
+    auto lref = ref->cast<LogicalCachedExtent>();
+    rewrite_block_list[lref->extent_writer].emplace_back(lref);
```

Comment:
I don't think rewrite_block_list makes much sense.  It seems to be used within the same rewrite_extents method in TransactionManager and for a small amount of manipulation in Cache::complete_commit.  I'd rather this logic removed from Cache and Transaction and dealt with at a higher level.